### PR TITLE
Added the unit test for mesheryctl system restart command

### LINKED ISSUE #3407: [mesheryctl] (child) `mesheryctl system` unit testing
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

check Epic here #3016 

#### Current Behavior
Currently, we don't have any unit tests setup to test `mesheryctl system` commands and functions

#### Desired Behavior
Required to enable unit testing support for `mesheryctl system` commands and functions, to ensure the accuracy and robustness of each mesheryctl release.

**mesheryctl Unit Testing**
- Mark all possible situations for each `mesheryctl system` commands and functions.
- Write tests using Golang’s standard library.
- A combination of CodeCov and GitHub Actions are to be used as mainstays in the approach to unit testing - https://github.com/codecov/codecov-action.

#### Subcommands to cover

- [x] mesheryctl system start
- [x] mesheryctl system stop
- [x] mesheryctl system context
- [x] mesheryctl system channel
- [x] mesheryctl system token
- [x] mesheryctl system logs
- [x] mesheryctl system reset
- [x] mesheryctl system restart
- [x] mesheryctl system status
- [x] mesheryctl system update
- [x] mesheryctl system config

---

#### Contributor Resources

- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Shreyas220**
Signed-off-by: Shreyas220 <shreyas.ny@gmail.com>

**Description**
Added the unit test for mesheryctl system restart command

This PR fixes #3407

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by sayantan1413 — CHANGES_REQUESTED** [review]
Hey @Shreyas220 I think we can add the test for the docker platform as well

**Comment by leecalcote** [comment]
Unit tests!! Yes, please. 🙂

**Comment by Shreyas220** [comment]
@piyushsingariya little help needed

File: mesheryctl/internal/cli/root/system/fixtures/.meshery/meshery.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+services:
```
Comment (piyushsingariya) [inline mesheryctl/internal/cli/root/system/fixtures/.meshery/meshery.yaml:1]: This file isn't necessary. or is it?

File: mesheryctl/internal/cli/root/system/system_test.go [hunk 2]
```diff
@@ -87,6 +87,21 @@ func TestSystemCmdIntegration(t *testing.T) {
...
+			ExpectError:     false,
+			TimeoutRequired: 0,
+		},
+		{
+			Name:            "Restart Meshery and skiping update",
+			Action:          "restart",
+			Args:            []string{"restart", "--skip-update"},
+			ExpectError:     false,
+			TimeoutRequired: 0,
+		},
```
Comment (piyushsingariya) [inline mesheryctl/internal/cli/root/system/system_test.go:104]: These should be shifted to appropriate places, like after the `system start` test for both docker and k8s.

File: mesheryctl/pkg/utils/TestConfig.yaml [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
+    token: Default
+    platform: docker
+    adapters:
+    - meshery-istio
+    channel: stable
+    version: v0.5.10
+current-context: local
+tokens:
+- location: auth.json
+  name: Default
```
Comment (piyushsingariya) [inline mesheryctl/pkg/utils/TestConfig.yaml]: reset this file, this file is being used by so many things I can't even count.

**Review by piyushsingariya — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/system/system_test.go [hunk 4]
```diff
@@ -59,27 +70,39 @@ func TestSystemCmdIntegration(t *testing.T) {
...
 			Action:          "start",
 			Args:            []string{"start", "-p", "kubernetes", "-y"},
 			ExpectError:     false,
 			TimeoutRequired: 1,
 		},
+		//restart
+		{
+			Name:            "Restart Meshery",
+			Action:          "restart",
+			Args:            []string{"restart kubernetes"},
```
[the remark this answers is not in the document]
Comment (Shreyas220) [reply mesheryctl/internal/cli/root/system/system_test.go:85]: oh sorry, i was trying to add it in Action

File: mesheryctl/internal/cli/root/system/system_test.go [hunk 5]
```diff
@@ -36,20 +36,31 @@ func TestSystemCmdIntegration(t *testing.T) {
...
 			Name:            "Start Meshery with Docker platform",
 			Action:          "start",
 			Args:            []string{"start", "-p", "docker", "-y"},
 			ExpectError:     false,
 			TimeoutRequired: 1,
 		},
+		//restart
+		{
+			Name:            "Restart Meshery",
+			Action:          "restart docker",
```
Comment (piyushsingariya) [inline mesheryctl/internal/cli/root/system/system_test.go:50]:
```suggestion
			Action:          "restart",
```

File: mesheryctl/internal/cli/root/system/system_test.go [hunk 6]
```diff
@@ -59,27 +70,39 @@ func TestSystemCmdIntegration(t *testing.T) {
...
 			Name:            "Start Meshery with Kubernetes platform",
 			Action:          "start",
 			Args:            []string{"start", "-p", "kubernetes", "-y"},
 			ExpectError:     false,
 			TimeoutRequired: 1,
 		},
+		//restart
+		{
+			Name:            "Restart Meshery",
+			Action:          "restart kubernetes",
```
Comment (piyushsingariya) [inline mesheryctl/internal/cli/root/system/system_test.go:84]:
```suggestion
			Action:          "restart",
```

**Review by piyushsingariya — CHANGES_REQUESTED** [review]
it's ready to merge after these changes

**Review by piyushsingariya — APPROVED** [review]

**Pull request merged**